tests/dosbox_test_fixture.h

```diff
@@ -0,0 +1,69 @@
+#ifndef DOSBOX_TEST_FIXTURE_H
+#define DOSBOX_TEST_FIXTURE_H
+
+#include <iterator>
+#include <string>
+
+#include <gtest/gtest.h>
+
+#define SDL_MAIN_HANDLED
+
+#include "control.h"
+
+class DOSBoxTestFixture : public ::testing::Test {
+public:
+	DOSBoxTestFixture()
+	        : arg_c_str("-conf tests/files/dosbox-staging-tests.conf\0"),
+	          argv{arg_c_str},
+	          com_line(1, argv),
+	          config(Config(&com_line))
+	{
+		control = &config;
+	}
+
+	void SetUp() override
+	{
+		// Create DOSBox Staging's config directory, which is a
+		// pre-requisite that's asserted during the Init process.
+		//
+		CROSS_DetermineConfigPaths();
+		const auto config_path = CROSS_GetPlatformConfigDir();
+		SETUP_ParseConfigFiles(config_path);
+
+		Section *_sec;
+		// This will register all the init functions, but won't run them
+		DOSBOX_Init();
+
+		for (auto section_name : sections) {
+			_sec = control->GetSection(section_name);
+			// NOTE: Some of the sections will return null pointers,
+			// if you add a section below, make sure to test for
+			// nullptr before executing early init.
+			_sec->ExecuteEarlyInit();
+		}
+
+		for (auto section_name : sections) {
+			_sec = control->GetSection(section_name);
+			_sec->ExecuteInit();
+		}
+	}
+
+	void TearDown() override
+	{
+		std::vector<std::string>::reverse_iterator r = sections.rbegin();
+		for (; r != sections.rend(); ++r)
```

Comment:
Can let the return-type of `.rbegin()` control `r`'s type.

`for (auto r = sections.rbegin(); r != sections.rend(); ++r)`

Then if the container format for `sections` is refactored (and that contaienr still support a reverse begin iterator), then it will follow suit here too without needing code updates.

Auto if often thought of as reckless or lazy (that's how I thought of it).  But when it's used carefully, as in "let the right-hand-side dictate the type', it's actually very powerful (allowing for easier refactoring) and lets the types become more exacting to the specification of the RHS.